src/dialogxml/widgets/button.cpp

```diff
@@ -164,13 +184,19 @@ void cButton::validatePostParse(ticpp::Element& elem, std::string fname, const s
 	if(labelledButtons.count(type)) {
 		if(!attrs.count("color") && !attrs.count("colour") && parent->getBg() == cDialog::BG_DARK)
 			setColour(sf::Color::White);
-		if(!getText().empty() && !attrs.count("width"))
+		if(type != BTN_TINY && type != BTN_LED && !getText().empty() && !attrs.count("width"))
```

Comment:
…wait. Why do buttons require width? Other than tiny, LED, and the red push button, isn't the width a fixed function of the type? I'm confused…

Reply:
Ah, I get it. This is in a block that is only reached if the button is one of those three types. So, removing the check entirely is probably correct after all (and also implement some kind of automatic size detection for `BTN_PUSH`).